sympy/solvers/solveset.py

```diff
@@ -863,15 +639,14 @@ def solveset(f, symbol=None, domain=S.Complexes):
         It is a bug, please report to the github issue tracker.
 
 
-    `solveset` uses two underlying functions `solveset_real` and
-    `solveset_complex` to solve equations. They are the solvers for real and
-    complex domain respectively. `solveset` ignores the assumptions on the
-    variable being solved for and instead, uses the `domain` parameter to
-    decide which solver to use.
-
     Notes
     =====
 
+    The output of `solveset` depends on the requested domain. It
+    ignores assumptions on the symbol procided by design: it will
+    always be clear what domain is being used without need to
```

Comment:
I modified this as

```
    The output of `solveset` depends on the requested domain rather
    than on the assumptions a symbol may have. Assumptions on symbols
    are ignored.
```

It's not really clearer since writing `x = symbol('x', real=True); solveset(foo, x)` is no more obfuscated than `dom = S.Reals; solveset(foo, x, dom)`. But the domain can be constructed from sets to be more precise than assumptions (e.g. you could set the domain to be any subset of reals).